src/http/client.cc

```diff
@@ -390,7 +390,14 @@ future<> client::do_make_request(connection& con, request& req, reply_handler& h
             });
         }
 
-        return handle(rep, con.in(rep)).finally([reply = std::move(reply)] {});
+        return handle(rep, con.in(rep)).finally([reply = std::move(reply), &con] {
+            if (reply->content_length != reply->consumed_content) {
+                http_log.trace("content was not fully consumed, content length is {} but consumed only {}, will close the connection",
+                               reply->content_length,
+                               reply->consumed_content);
+                con._persistent = false;
```

Comment:
nitpicking: This marking and log message is not really needed in case we get here by exception

Reply:
how about now? moved the code from `finally` to `then` and then the exception, I guess, will propagate to outer `handle_exception`